jupyterlab-ihaskell/src/codemirror-ihaskell.ts

```diff
@@ -1,31 +1,36 @@
-import * as CodeMirror from 'codemirror';
+// @ts-nocheck
+// The types for CodeMirror don't accept a RegExp for 'open' and 'close'
 
-import 'codemirror/lib/codemirror';
-import 'codemirror/mode/haskell/haskell';
+import { ICodeMirror, Mode } from '@jupyterlab/codemirror';
 
-CodeMirror.defineMode("ihaskell", (config) => {
-  let hmode = CodeMirror.getMode(config, "haskell");
-  return CodeMirror.multiplexingMode(
-    hmode,
-    {
-      open: /:(?=!)/, // Matches : followed by !, but doesn't consume !
-      close: /^(?!!)/, // Matches start of line not followed by !, doesn't consume character
-      mode: CodeMirror.getMode(config, "text/plain"),
-      delimStyle: "delimit"
-    },
-    {
-      open: /\[r\||\[rprint\||\[rgraph\|/,
-      close: /\|\]/ ,
-      mode: CodeMirror.getMode(config, "text/x-rsrc"),
-      delimStyle: "delimit"}
-  );
-});
+export function defineIHaskellMode({ CodeMirror }: ICodeMirror) {
+  Mode.ensure("haskell");
```

Comment:
hm, these are promises... might be a little racy, but you're _probably_ fine...

Reply:
How would you suggest doing it instead?

Reply:
`export async function defineIHaskellMode` and then `await Mode.ensure` in both places. Then the index.ts could `defineIHaskellMode(codeMirror).catch(console.warn)` (without making  _the_ calling function async).

Reply:
Fixed in https://github.com/gibiansky/IHaskell/commit/92027da0f3486525d07fd462e19bbdd44dc9218f, thanks!